packages/ui-library/src/components/icon-menu/index.tsx

```diff
@@ -63,16 +99,40 @@ const IconMenu = ({ options, color = 'info', variant = 'text', icon, size = 'sma
           horizontal: 'left',
         }}
       >
-        {options.map(({ onClick, icon: itemIcon, label, control, closeOnClick = true }) => (
-          <MenuItem key={label} onClick={() => handleCloseWithAction(closeOnClick, onClick)}>
-            {itemIcon}
-            {label}
-            {control}
-          </MenuItem>
-        ))}
+        {options.map((option, index) => {
+          if (option.type === IconMenuOptionType.option) {
+            const {
+              label,
+              closeOnClick = true,
+              color: itemColor,
+              control,
+              onClick,
+              icon: itemIcon,
+              secondaryLabel,
+            } = option;
+            return (
+              <MenuItem
+                onClick={() => handleCloseWithAction(closeOnClick, onClick)}
+                style={{ color: itemColor }}
```

Comment:
style should not be used on MUI components, they accept an `sx` property. In any case this should be a ButtonProps['color'] right?

Reply:
I'm changing the prop applied! Thanks 
About ButtonProps, not all of them are applicable to Typography (color=success is not for instance)

Reply:
We could still send it as the buttonProp and inside the component use a map for the button

Reply:
I found a way to apply most of the button color props to the Typography component
https://github.com/mui/material-ui/issues/29564

I think it's good to go for our current use cases